src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java

```diff
@@ -310,11 +314,9 @@ public List<BrAPIGermplasm> createBrAPIGermplasm(List<BrAPIGermplasm> postBrAPIG
         var program = programDAO.fetchOneById(programId);
         try {
             if (!postBrAPIGermplasmList.isEmpty()) {
-                Callable<Map<String, BrAPIGermplasm>> postFunction = () -> {
                     List<BrAPIGermplasm> postResponse = brAPIDAOUtil.post(postBrAPIGermplasmList, upload, api::germplasmPost, importDAO::update);
-                    return processGermplasmForDisplay(postResponse, program.getKey());
-                };
-                return programGermplasmCache.post(programId, postFunction);
+                    processGermplasmForDisplay(postResponse, program.getKey());
+                    return postResponse;
```

Comment:
```suggestion
            return new ArrayList(processGermplasmForDisplay(postResponse, program.getKey()).values());
```

I recommend making the above change or similar to preserver the current behavior.

Reply:
Done.